go/libraries/doltcore/schema/typeinfo/decimal.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package typeinfo
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/src-d/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+const (
+	decimalTypeParam_Precision = "prec"
+	decimalTypeParam_Scale     = "scale"
+)
+
+// This is a dolt implementation of the MySQL type Decimal, thus most of the functionality
+// within is directly reliant on the go-mysql-server implementation.
+type decimalType struct {
+	sqlDecimalType sql.DecimalType
+}
+
+var _ TypeInfo = (*decimalType)(nil)
+
+func CreateDecimalTypeFromParams(params map[string]string) (TypeInfo, error) {
+	var precision uint8
+	if precisionStr, ok := params[decimalTypeParam_Precision]; ok {
+		precisionUint, err := strconv.ParseUint(precisionStr, 10, 8)
+		if err != nil {
+			return nil, err
+		}
+		precision = uint8(precisionUint)
+	} else {
+		return nil, fmt.Errorf(`create decimal type info is missing param "%v"`, decimalTypeParam_Precision)
+	}
+	var scale uint8
+	if scaleStr, ok := params[decimalTypeParam_Scale]; ok {
+		scaleUint, err := strconv.ParseUint(scaleStr, 10, 8)
+		if err != nil {
+			return nil, err
+		}
+		scale = uint8(scaleUint)
+	} else {
+		return nil, fmt.Errorf(`create decimal type info is missing param "%v"`, decimalTypeParam_Scale)
+	}
+	sqlDecimalType, err := sql.CreateDecimalType(precision, scale)
+	if err != nil {
+		return nil, err
+	}
+	return &decimalType{sqlDecimalType}, nil
+}
+
+// ConvertNomsValueToValue implements TypeInfo interface.
+func (ti *decimalType) ConvertNomsValueToValue(v types.Value) (interface{}, error) {
+	if val, ok := v.(types.String); ok {
```

Comment:
We can't store numeric types as strings because they won't sort properly, which matters a lot for efficient lookups.

I would sabotage this type to make all methods panic until we have a conversation about how to store these in noms.